### ISSUE TITLE
[Feature] Generate the post body for a form widget automatically

### ISSUE THREAD

**Issue by areyabhishek** [appsmithorg/appsmith#2543]
## Summary

It's difficult to post data from a form widget with many fields. It's easy to get the fields wrong when creating the post body using auto-complete. Can we generate the post body for an API automatically?

**Comment by Nikhil-Nandagopal** [comment]
@areyabhishek we generate the values of the form today but it's not possible to generate the value of the post body because we don't know the keys that the API accepts. So right now {{Form1.data}} gives the value of all the elements in the form but this won't be sufficient. Any other ideas here?

**Comment by areyabhishek** [comment]
How about generating the template so that the user can modify it?

**Comment by Nikhil-Nandagopal** [comment]
@areyabhishek was thinking about a solution for this. What do you think about doing this the opposite way, i.e generate the form from the API instead of the API from the form? This way we can actually guarantee some level of success because every developer should have the CURL of the submit API.

**Comment by areyabhishek** [comment]
Yes, that would be pretty cool. The user can choose the fields+type for every input.
